cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandlerInternal.java

```diff
@@ -200,4 +210,41 @@ public void getAppDetailForVersion(HttpRequest request, HttpResponder responder,
         : applicationLifecycleService.getAppDetail(appId);
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(appDetail));
   }
+
+  /**
+   * Update git metadata of the provided applications.
+   *
+   * @param request {@link FullHttpRequest}
+   * @param responder {@link HttpResponse}
+   * @param namespace the namespace where the applications are deployed
+   * @throws Exception if either namespace or any of the application doesn't exist
+   */
+  @POST
+  @Path("/apps/updateGitMetadata")
+  public void updateGitMetadata(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") final String namespace) throws Exception {
+    NamespaceId namespaceId = new NamespaceId(namespace);
+    if (!namespaceQueryAdmin.exists(namespaceId)) {
+      throw new NamespaceNotFoundException(namespaceId);
+    }
+    Type listOfAppUpdateGitMetaRequestType = new TypeToken<ArrayList<AppUpdateGitMetaRequest>>(){}.getType();
+    List<AppUpdateGitMetaRequest> appsUpdateGitMetaRequest;
+
+    try {
+      appsUpdateGitMetaRequest = parseBody(request, listOfAppUpdateGitMetaRequestType);
```

Comment:
In general, request bodies should be objects and not collections, even if it is just an object with a single collection field. This allows the API to be more extensible. For example, in the future we could add an options field or something like that.

(This is not always followed in some of the older APIs, but those are bad examples).

Reply:
done.